### PR TITLE
Enable im2col col2im

### PR DESCRIPTION
**Pull request by ZhiweiYan-96**
# Description
Enable operator `im2col`, `col2im`, add simple cases

### CODE REVIEW

**Comment by ZhiweiYan-96** [comment]
@fengyuan14  @EikanWang  @xytintel  Could you please help review the PR? Thanks!

File: src/comm/SYCLHelpers.h [hunk 1]
```diff
@@ -119,6 +118,34 @@ static inline void sycl_kernel_submit(
   q.submit(cgf);
 }
 
+template <typename ker_t>
+static inline void sycl_kernel_submit(
+    int64_t global_range,
+    ::sycl::queue q,
```
Comment (gujinghui) [inline src/comm/SYCLHelpers.h:124]: should be a reference?

File: src/comm/SYCLHelpers.h [hunk 2]
```diff
@@ -119,6 +118,34 @@ static inline void sycl_kernel_submit(
...
+    cgh.parallel_for<ker_t>(::sycl::range<1>(global_range), ker);
+  };
+  // XXX: c10::xpu::getStreamFromPool().queue();
+  q.submit(cgf);
+}
+
+template <typename ker_t, typename ker_creator_t>
+static inline void sycl_kernel_submit(
+    int64_t global_range,
+    ::sycl::queue q,
```
Comment (gujinghui) [inline src/comm/SYCLHelpers.h:136]: ditto

File: src/aten/sycl/Col2ImKernel.cpp [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+      pad_w,
+      stride_h,
+      stride_w,
+      dilation_h,
+      dilation_w,
+      out_data);
+  sycl_kernel_submit(total_threads, sycl_queue, kfn);
+}
+
+void col2im_out_template(
```
Comment (gujinghui) [inline src/aten/sycl/Col2ImKernel.cpp:140]: Do we consider to refine the cuda/cpu templates, instead of copy out?

**Comment by EikanWang** [comment]
@ZhiweiYan-96 , may I know if you have addressed Jinghui's comments?

**Comment by AlienLiang23** [comment]
image
image

all UT cases passed

File: src/aten/sycl/Col2ImKernel.cpp (hunk 3 above)
Comment (AlienLiang23) [reply src/aten/sycl/Col2ImKernel.cpp:140]: Sure. We can use **#include <ATen/native/im2col_shape_check.h>** instead of current.

**Comment by fengyuan14** [comment]
Please paste cases involve nn.functional.unfold, im2col, and col2im. unfold is a separate op.

**Comment by fengyuan14** [comment]
Fine gran cases:
image

**Pull request merged**